pkg/endpoint/policy.go

```diff
@@ -498,6 +499,230 @@ func (e *Endpoint) IngressOrEgressIsEnforced() bool {
+					log.Debugf("Error while getting port number for %s", port)
+					continue
+				}
+				l4RuleCtx := policy.L4RuleContext{
+					Port: port,
+					// FIXME do the translation of protocols to uint8
+					// For now 0 means any
+					Proto: 0,
+				}
+				if _, ok := l4Ports[l4RuleCtx]; !ok {
+					l4Ports[l4RuleCtx] = policy.L7RuleContext{}
+				}
+			}
+			// FIXME When/How to store the ProxyPort?
+		}
+
+		// FIXME We need a list of all k8s services so we can match them
+		// by labels / name
+		// The ToPorts should be enforced the policy in the Service Port and not
+		// in the endpoint port that is backed by the Service.
+
+		//for srcID, srcLabels := range *labelsMap {
+		//	for _, es := range eg.ToServices {
+		//		if es.Matches(srcLabels) {
+		//			if _, ok := ingressPolicy.ByNumericIdentity[srcID]; !ok {
+		//				egressPolicy.ByNumericIdentity[srcID] = l4Ports
+		//			}
+		//		}
+		//	}
+		//}
+
+		// Create the policy ByEntity
+		for _, entity := range eg.ToEntities {
+			egressPolicy.ByEntity[entity] = l4Ports
+		}
+
+		// Create the policy ByCIDR
+		for _, cidr := range eg.ToCIDR {
+			egressPolicy.ByCIDR[cidr] = l4Ports
+		}
+	}
+
+	// No errors so do atomic assignment
+	e.IngressPolicy = ingressPolicy
+	e.EgressPolicy = egressPolicy
+	// Q: When to bump policy revision?
+	// A: When bpf enforces it
+	// But we still need to store the policy revision somewhere
+	e.nextPolicyRevision = rev
+
+	return oldIngressPolicy, oldEgressPolicy, oldPolicyRev
+}
+
+func (e *Endpoint) DebugPolicy() {
+	ingressPolicy := e.IngressPolicy
+	egressPolicy := e.EgressPolicy
+
+	// At this point we should have the policies calculated except the proxy
+	// Port. The proxy port creation/update should be done in a new function
+	// called after regeneratePolicyNew
+
+	// Ok, how to read the policy generated?
+	// For ingressPolicy.ByNumericIdentity
+	for ni, l4Rules := range ingressPolicy.ByNumericIdentity {
+		if l4Rules.IsL3Only() {
+			log.Debugf("EP %s will accept receiving NumericIdentity %s on all L4 ports", e.ID, ni)
+			//+-----------------------+----------------------------+
+			//|         KEY           |            VALUE           |
+			//+-IDENTITY-+-PORT/PROTO-+-REDIRECT-+-BYTES-+-PACKETS-+
+			//|    ni    |      0     |    0     |   0   |    0    |
+			//+-----------------------+----------------------------+
+		} else {
+			for l4Policy, l7Policy := range l4Rules {
+				pp := l4Policy.PortProto()
+				redi := l7Policy.RedirectPort
+				log.Debugf("EP %s will accept receiving NumericIdentity %s with PortProto %s redirects to %d", e.ID, ni, pp, redi)
+				//+-----------------------+----------------------------+
+				//|         KEY           |            VALUE           |
+				//+-IDENTITY-+-PORT/PROTO-+-REDIRECT-+-BYTES-+-PACKETS-+
+				//|    ni    |     pp     |   redi   |   0   |    0    |
+				//+-----------------------+----------------------------+
+			}
+		}
+	}
+
+	// I haven't figure out how BPF will map the FromCIDR and FromEntities
```

Comment:
Is there any reason `FromEntities` can't fold into the same policy map? We restrict the reserved identities anyway.

For `FromCIDR` I expect it to remain as a separate map with similar lookups to what the policy has, in terms of performing L3+L4 lookup first, then try L3-only, then L4-only. The main difference is that it's also an LPM.

Reply:
>Is there any reason FromEntities can't fold into the same policy map? We restrict the reserved identities anyway.

Ah you're right. So `FromEntities` map can actually be merged into the `ByNumericIdentity`